File: dsa-learning/dsa-java-code/src/main/java/com/dsa/leetcode/labuladong/array/prefix_sum/P303_RangeSumQuery_Immutable.java
package com.dsa.leetcode.labuladong.array.prefix_sum;

public class P303_RangeSumQuery_Immutable {
    int[] sums;
    public P303_RangeSumQuery_Immutable(int[] nums) {
        int n = nums.length;
        sums = new int[n + 1];
        for (int i = 0; i < n; i++) {
            sums[i + 1] = sums[i] + nums[i];
        }
    }

    public int sumRange(int left, int right) {
        return sums[right + 1] - sums[left];
    }
}

File: dsa-learning/dsa-python-code/com/dsa/leetcode/labuladong/array/prefix_sum/P303_RangeSumQuery_Immutable.py
from typing import List


class NumArray:

    def __init__(self, nums: List[int]):
        self.sums = [0]
        sums = self.sums
        for num in nums:
            sums.append(sums[-1] + num)

    def sumRange(self, left: int, right: int) -> int:
        return self.sums[right + 1] - self.sums[left]
